### ISSUE TITLE
Follower counter on newly reinstated accounts is negative rather than zero

### ISSUE THREAD

**Issue by kaniini** [mastodon/mastodon#18548]
### Steps to reproduce the problem

I had a user on my instance decide to delete their account.  After a few weeks he changed his mind and decided he wanted it reinstated, so I used tootctl to do it based on [this discourse thread I saw on the topic](https://discourse.joinmastodon.org/t/is-it-possible-to-restore-a-deleted-user-account/1945):

```
kaniini@penelo ~ % kubectl exec --stdin --tty service/mastodon-web -- bundle exec tootctl accounts create kddlb --reattach --force --email=[elided]
New password: [elided]
```

### Expected behaviour

The follower/following counters should be zeroed when the account is reinstated.

### Actual behaviour

The follower/following counters are negative.

### Specifications

Mastodon 3.5.2+glitch, but I am certain it is unrelated to the glitch patchset.

**Comment by kode54** [comment]
Maybe the issue is a race condition in the deletion code before it leaves an empty profile behind. For example, maybe it zeroes the counters, then the code which blanks the followers/follows lists kicks in and decrements the counter as it's deleting the lists.

**Comment by kaniini** [comment]
Yes, my guess is that something happened with Sidekiq that put it in this state but `tootctl` could zero the counters when the account is reinstated.

**Comment by trwnh** [comment]
Possibly related to #18365
